src/validation.cpp

```diff
@@ -2327,7 +2327,7 @@ bool CChainState::FlushStateToDisk(
         // Flush best chain related state. This can only be done if the blocks / block index write was also done.
         if (fDoFullFlush && !CoinsTip().GetBestBlock().IsNull()) {
             LOG_TIME_SECONDS(strprintf("write coins cache to disk (%d coins, %.2fkB)",
-                coins_count, coins_mem_usage / 1000));
+                coins_count, coins_mem_usage / 1000), BCLog::ALL);
```

Comment:
I don't really like leaking this private implementation detail of the log flags into the caller code. I'd slightly prefer a type-safe way, that also better documents what is happening.

What about making this argument optional not with VA_ARGS, but with an `Optional<BCLog::LogFlags>`? Passing a nullopt or `{}` will unconditionally log, whereas passing a log flag will only log to that category (if enabled).

Reply:
We could drop it altogether as this is the only call to `LOG_TIME_SECONDS`:

```diff
diff --git a/src/logging.h b/src/logging.h
index 5d8ec33392..b2fde1b9ea 100644
--- a/src/logging.h
+++ b/src/logging.h
@@ -178,7 +178,7 @@ static inline void LogPrintf(const char* fmt, const Args&... args)
 // evaluating arguments when logging for the category is not enabled.
 #define LogPrint(category, ...)              \
     do {                                     \
-        if ((category) == BCLog::LogFlags::ALL || LogAcceptCategory((category))) { \
+        if (LogAcceptCategory((category))) { \
             LogPrintf(__VA_ARGS__);          \
         }                                    \
     } while (0)
diff --git a/src/logging/timer.h b/src/logging/timer.h
index 34a6e25185..6a40dc48ba 100644
--- a/src/logging/timer.h
+++ b/src/logging/timer.h
@@ -95,8 +95,8 @@ private:
 
 #define LOG_TIME_MILLIS(end_msg, log_category) \
     BCLog::Timer<std::chrono::milliseconds> PASTE2(logging_timer, __COUNTER__)(__func__, end_msg, log_category)
-#define LOG_TIME_SECONDS(end_msg, log_category) \
-    BCLog::Timer<std::chrono::seconds> PASTE2(logging_timer, __COUNTER__)(__func__, end_msg, log_category)
+#define LOG_TIME_SECONDS(end_msg) \
+    BCLog::Timer<std::chrono::seconds> PASTE2(logging_timer, __COUNTER__)(__func__, end_msg)
 
 
 #endif // BITCOIN_LOGGING_TIMER_H
diff --git a/src/validation.cpp b/src/validation.cpp
index a0ac003029..bab04b8e34 100644
--- a/src/validation.cpp
+++ b/src/validation.cpp
@@ -2328,7 +2328,7 @@ bool CChainState::FlushStateToDisk(
         // Flush best chain related state. This can only be done if the blocks / block index write was also done.
         if (fDoFullFlush && !CoinsTip().GetBestBlock().IsNull()) {
             LOG_TIME_SECONDS(strprintf("write coins cache to disk (%d coins, %.2fkB)",
-                coins_count, coins_mem_usage / 1000), BCLog::ALL);
+                coins_count, coins_mem_usage / 1000));
 
             // Typical Coin structures on disk are around 48 bytes in size.
             // Pushing a new one to the database can cause it to be written
```

Reply:
Updated this branch: https://github.com/fanquake/bitcoin/compare/no_gnu_extensions...Empact:2020-02-no_gnu_extensions

Reply:
> We could drop it altogether

I've updated to do this.